core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -57,7 +56,8 @@
     enum State {
         NEEDS_TO_WRITE_FIRST_HEADER,
         NEEDS_DATA_OR_TRAILING_HEADERS,
-        DONE
+        DONE_1,
+        DONE_2
```

Comment:
I think we can remove `DONE_2`. This is needed to prevent calling `subscription.cancel()` twice, but according to the [Reactive Streams JVM spec 3.5](https://github.com/reactive-streams/reactive-streams-jvm#3.5), `subscription.cancel()` must be idempotent, so it doesn't matter.

Reply:
On second thought after reading your comment, it would be better to remove `DONE_2` because the situation I wanted to prevent is rare. Thanks!